### PR TITLE
fix(webapp/sidenav): Move Browse sidebar section in between Trash and other collections

### PR DESCRIPTION
**Pull request by rafpaf**
### Description

Moves the **Browse** part of the sidenav down, just above Trash.

Brief discussion here: https://metaboat.slack.com/archives/C06REDHM6BV/p1722460903919469?thread_ts=1720458369.496439&cid=C06REDHM6BV

### How to verify

Open the sidenav

### CODE REVIEW

**Comment by rafpaf** [comment]
* **#46457** <a href="https://app.graphite.dev/github/pr/metabase/metabase/46457?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rafpaf and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by dhuniverse** [comment]
Approved.

**Review by dhuniverse — APPROVED** [review]
Looks good to me. Approved.

File: frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarView.tsx [hunk 1]
```diff
@@ -120,6 +121,19 @@ function MainNavbarView({
...
+    let trashCollection;
+    const collectionsWithoutTrash = [];
+    for (const c of collections) {
+      if (c.type === "trash") {
+        trashCollection = c;
+      } else {
+        collectionsWithoutTrash.push(c);
+      }
+    }
+    return [collectionsWithoutTrash, trashCollection];
```
Comment (iethree) [inline frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarView.tsx:134]:
```suggestion
    return _.partition(collections, (c) => c.type !== "trash");
```

and the trash will already be in an array so you can pass it straight to the tree component below

**Review by iethree — APPROVED** [review]
👍

**Review by iethree — APPROVED** [review]
❤️

**Pull request merged**